web-ui/jsoo/main.ml

```diff
@@ -0,0 +1,80 @@
+module Dom_html = Js_of_ocaml.Dom_html
+module Dom = Js_of_ocaml.Dom
+module Js = Js_of_ocaml.Js
+module Firebug = Js_of_ocaml.Firebug
+
+(* module WebSockets = Js_of_ocaml.WebSockets *)
+module WebSocket = Brr_io.Websocket
+module Ev = Brr.Ev
+
+let regexp_left_paren = Re.Str.regexp_string "("
+let regexp_right_paren = Re.Str.regexp_string ")"
+let encoded_left_paren = "%28"
+let encoded_right_paren = "%29"
+
+let encode_parens s : Jstr.t =
+  let s' = Jstr.to_string s in
+  let s_lp_encoded =
+    Re.Str.global_replace regexp_left_paren encoded_left_paren s'
+  in
+  let s_rp_encoded =
+    Re.Str.global_replace regexp_right_paren encoded_right_paren s_lp_encoded
+  in
+  Jstr.of_string s_rp_encoded
+
+(* let js_str = Js.string *)
+let document = Dom_html.document
```

Comment:
It might be worthwhile going _all in_ on `Brr` so you can drop the dependency on `js_of_ocaml` and `js_of_ocaml-ppx`? In which case this could turn into something like:

```ocaml
let inject_log_lines data =
  match Document.find_el_by_id G.document scroller_id with
  | None -> assert false
  | Some scroller ->
    let new_div = El.div [ El.txt' data ] in
    El.append_children scroller [ new_div ]
```

Reply:
Thanks @patricoferris I had intended to get us to just using Brr but since I am still quite unfamiliar with the API, had deferred completing that. This will help a lot 🙂

Reply:
Just using `Brr` looks like a good option for making this code simpler.